### ISSUE TITLE
Differing Accuracy Results: Torchmetrics vs. Sklearn on Small Sample

### ISSUE THREAD

**Issue by NorbertRop** [Lightning-AI/torchmetrics#2312]
## 🐛 Bug

I'm encountering an issue while calculating accuracy for a task. The results differ slightly between using torchmetrics and sklearn's `accuracy_score`.

### To Reproduce

[results.json](https://github.com/Lightning-AI/torchmetrics/files/13965391/results.json)

```py
python
import json
import torch
from torchmetrics import Accuracy
from torchmetrics.functional import accuracy
from sklearn.metrics import accuracy_score

with open('results.json') as f:
    data = json.load(f)
    targets = torch.tensor(data['targets'])
    preds = torch.tensor(data['preds'])

    acc = Accuracy(task='multiclass', num_classes=len(set(targets)))
    acc(preds, targets)
    print(acc.compute().item())
    print(accuracy(preds, targets, task='multiclass', num_classes=len(set(targets))).item())
    print(accuracy_score(targets, preds))
 ```

Output
```
0.9599999785423279
0.9599999785423279
0.96
```

The discrepancy is very small, but it's puzzling, especially with a small sample size like 50.

### Expected behavior

The same output for torchmetrics and sklearn.

### Environment
```
torchmetrics == 1.2.1
scikit-learn==1.3.2
torch==2.1.2
python==3.11.5
```

**Comment by SkafteNicki** [comment]
Hi @NorbertRop, thanks for reporting this and sorry for the slow reply from my side.
What you are encountering here seems to be simple a case of limitations of floating-point arithmetic, which is caused due to the limited precision of floating-point numbers. If you had simply printed the tensor e.g.
```python
print(acc.compute())
```
you would get `tensor(0.9600)`. When calling `.item` you essentially call `float(acc.compute())` which for some reason converts it into a python native float which is `0.9599999785423279` and not `0.96`. Again, it is kind of similar to
```python
0.1 + 0.2
```
does not equal `0.3` in Python, instead it equals `0.30000000000000004`. 

How `.item` method behaves we really cannot do anything about. That has to do with Python + Pytorch.
Closing issue.
